Sync/Synchronizers/Bookmarks/BookmarksSynchronizer.swift

```diff
@@ -130,8 +130,66 @@ open class BufferingBookmarksSynchronizer: TimestampedSingleCollectionSynchroniz
                         return deferMaybe(result)
                     }
                 }
-
                 return deferMaybe(result)
+            } >>== { result in
+                guard case .completed = result else {
+                    return deferMaybe(result)
+                }
+                return storage.getLocalBookmarksAdditions() >>== { newBookmarks -> Success in
```

Comment:
As mentioned earlier, if `newBookmarks` is an array, and you add

```
let newBookmarkGUIDs = newBookmarks.map { $0.guid }
```

you can kill `newChildren` later, and skip a bunch of work in several places.